server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java

```diff
@@ -0,0 +1,338 @@
+        appendMessage("eml/twoAttachmentsTextPlain.eml");
+    }
+
+    @Given("^the user has a multipart message in \"([^\"]*)\" mailbox$")
+    public void appendMultipartMessageWithOneAttachments(String arg1) throws Throwable {
+        appendMessage("eml/htmlAndTextMultipartWithOneAttachment.eml");
+    }
+
+    private void appendMessage(String emlFileName) throws Exception {
+        ZonedDateTime dateTime = ZonedDateTime.parse("2014-10-30T14:12:00Z");
+        mainStepdefs.jmapServer.serverProbe().appendMessage(userStepdefs.username, 
+                new MailboxPath(MailboxConstants.USER_NAMESPACE, userStepdefs.username, "inbox"),
+                ClassLoader.getSystemResourceAsStream(emlFileName), 
+                Date.from(dateTime.toInstant()), false, new Flags());
+    }
+
+    @When("^the user is getting his messages and gives a non null accountId$")
+    public void postWithAccountId() throws Exception {
+        post("[[\"getMessages\", {\"accountId\": \"1\"}, \"#0\"]]");
+    }
+
+    @When("^the user is getting his messages and gives unknown arguments$")
+    public void postWithUnknownArguments() throws Exception {
+        post("[[\"getMessages\", {\"WAT\": true}, \"#0\"]]");
+    }
+
+    @When("^the user is getting his messages and gives invalid argument$")
+    public void postWithInvalidArguments() throws Exception {
+        post("[[\"getMessages\", {\"ids\": null}, \"#0\"]]");
+    }
+
+    @When("^the user is getting his messages$")
+    public void post() throws Throwable {
+        post("[[\"getMessages\", {\"ids\": []}, \"#0\"]]");
+    }
+
+    @When("^the user is getting his messages and gives a list of ids \"(.*?)\"$")
+    public void postWithAListOfIds(String ids) throws Throwable {
+        post("[[\"getMessages\", {\"ids\": " + ids + "}, \"#0\"]]");
+    }
+
+    @When("^the user is getting his messages with parameters$")
+    public void postWithParameters(DataTable parameters) throws Throwable {
+        String payload = 
+                parameters.asMap(String.class, String.class)
+                    .entrySet()
+                    .stream()
+                    .map(this::entryToPair)
+                    .map(this::quoteIndex)
+                    .map(this::joinKeyValue)
+                    .collect(Collectors.joining(",", "{", "}"));
+        
+        post("[[\"getMessages\", " + payload + ", \"#0\"]]");
+    }
+
+    private Pair<String, String> entryToPair(Map.Entry<String, String> entry) {
+        return Pair.with(entry.getKey(), entry.getValue());
+    }
+
+    private Pair<String, String> quoteIndex(Pair<String, String> pair) {
+        return Pair.with(String.format("\"%s\"", pair.getValue0()), pair.getValue1());
+    }
+
+    private String joinKeyValue(Pair<String, String> pair) {
+        return Joiner.on(": ").join(pair);
+    }
+
+    private void post(String requestBody) {
+        post = with()
+            .accept(ContentType.JSON)
```

Comment:
You can configure defaults with : RestAssured.requestSpecification = blabla.contentType(blabla).accept(blabla)